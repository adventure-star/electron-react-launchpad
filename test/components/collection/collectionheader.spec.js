import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import CollectionHeader from '../../../src/renderer/components/collection/collectionheader'

function setupStandard() {
    const renderer = ReactTestRenderer.create(<CollectionHeader className='testClass'>Test collection content</CollectionHeader>);
    const component = renderer.toJSON();

    return {
        component
    };
}

describe('CollectionHeader component', () => {
    it('should be of type li', () => {
        const { component } = setupStandard();
        expect(component.type).toBe('li');
    });

    it('should have only the classes "collection-header" and "testClass"', () => {
        const { component } = setupStandard();
        expect(component.props.className).toBe('collection-header testClass');
    });

    it('first child should be h4', () => {
        const { component } = setupStandard();
        expect(component.children[0].type).toBe('h4');
    });

    it('children should be the text "Test collection content"', () => {
        const { component } = setupStandard();
        expect(component.children[0].children[0]).toBe('Test collection content');
    });

    it('Matches/Renders a snapshot', () => {
        const { component } = setupStandard();
        expect(component).toMatchSnapshot();
    });
});