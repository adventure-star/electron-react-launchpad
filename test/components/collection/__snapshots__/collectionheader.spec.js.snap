// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionHeader component Matches/Renders a snapshot 1`] = `
<li
  className="collection-header testClass"
  onClick={undefined}
>
  <h4>
    Test collection content
  </h4>
</li>
`;
